std/split/split_gl.asm

```diff
@@ -1,27 +1,25 @@
 // Splits an arbitrary field element into two u32s, on the Goldilocks field.
-machine SplitGL(RESET, operation_id) {
+machine SplitGL(RESET, _) {
 
-    operation split<0> in_acc -> output_low, output_high;
+    operation split in_acc -> output_low, output_high;
 
     // Latch and operation ID
     col fixed RESET(i) { i % 8 == 7 };
-    col witness operation_id;
 
     // 1. Decompose the input into bytes
 
     // The byte decomposition of the input, in little-endian order
     // and shifted forward by one (to use the last row of the
     // previous block)
-    col witness bytes;
-    // Puts the bytes together to form the input
+    col witness bytes(i) query ("hint", (in_acc' >> (((i + 1) % 8) * 8)) % 0xff);
```

Comment:
wat

Reply:
Too confusing? 😄 

- `((i + 1) % 8)` is the index in the current block, shifted by 1
- `(in_acc' >> (((i + 1) % 8) * 8))` is the input shifted by that many bytes
- `(in_acc' >> (((i + 1) % 8) * 8)) % 0xff)` selects the least significant byte

So in the end, it does what the comment above says that it should, but let me know if you have an idea how to make it clearer :)

Reply:
Yea it's fine. If the prover passes malicious values now, witgen won't fail, right?

Reply:
I'd phrase it differently, if the constraints allow for more than one value, witgen won't fail anymore. For example, if we remove step 3 and fix #593. That's why I think it's always better to rely on automatic witgen, but in this case it's not possible.